test/model.test.js

```diff
@@ -3826,6 +3826,57 @@ describe('Model', function() {
         });
       });
     });
+    it('is saved object with proper defaults', function(done) {
+      const DefaultTestObject = db.model('Test',
+        new Schema({
+          foo: {
+            x: { type: String },
+            y: { type: String }
+          },
+          boo: {
+            x: { type: Boolean, default: false }
+          },
+          bee: {
+            x: { type: Boolean, default: false },
+            y: { type: Boolean, default: false }
+          }
+        })
+      );
+      const myTest = new DefaultTestObject({
+        foo: { x: 'a', y: 'b' },
+        bee: {},
+        boo: {}
+      });
+
+      myTest.save(function(err, doc) {
```

Comment:
Using callback is considered old-fashioned, a more maintainable option would be to use `co/yield`, [here](https://github.com/Automattic/mongoose/blob/c01685a56eccca995c6880dcc6ea51c71d4dd43f/test/document.test.js#L9076)'s an example of how we use it in tests.

Reply:
Hello thanks for the comment, this pr created to demonstrate the issue https://github.com/Automattic/mongoose/issues/10441,